### PR TITLE
Bug: auth issues on reset password page just send the user home or to login without any message about an error

### LINKED ISSUE #465: Auth issues on reset password page just send the user home or to login without any message about an error
If there's an auth issue when a user tries to reset their password, the response will get caught in the client hooks and the user will be sent home (or to login depending on some variables), however the user will not be told that the reset failed, which you might guess if you end up at the login page, but if you are sent home you will just assume it succeed. The user should know that the reset failed.

I found this due to an auth bug, but we should improve the frontend experience as there's multiple reasons they would get an auth error on the reset password page.

### PR DESCRIPTION
**Pull request by myieye**
Resolves #465 

Here's what happens now if you get a 403 when trying to reset your password:
image

I also tried to show a message in cases where we do end up redirecting users to the login screen, but I got stuck when I discovered that we can't pass query-params when we `throw redirect(...)`, because whatever we pass in as a url gets URL-encoded.

@hahn-kev what do you think?

### CODE REVIEW

File: frontend/src/app.d.ts [hunk 1]
```diff
@@ -10,6 +10,14 @@ declare global {
     lexbox: { fetchProxy: Fetch }
   }
 
+  type LexboxResponseHandlingConfig = {
+    disabledStatusCodes?: number[];
```
Comment (hahn-kev) [inline frontend/src/app.d.ts:14]: I'd rather be more explicit about what we are using this for. To use this property correctly you need to know where it's consumed and what it's actually doing.
I'd rather something like `disableRedirectOnAuthError` or `ignoreAuthErrors` which is just a boolean property.

File: frontend/src/hooks.shared.ts [hunk 2]
```diff
@@ -20,7 +20,12 @@ export function getErrorMessage(error: unknown): string {
   );
 }
 
-export function validateFetchResponse(response: Response, isAtLogin: boolean, isHome: boolean): void {
+export function validateFetchResponse(
+  response: Response, isAtLogin: boolean, isHome: boolean, config?: LexboxResponseHandlingConfig): void {
```
Comment (hahn-kev) [inline frontend/src/hooks.shared.ts:24]: we could probably remove `isAtLogin` and use the new configuration property for the login request in user.ts

**Review by hahn-kev — CHANGES_REQUESTED** [review]
Looks good, but I'm not a fan of `disabledStatusCodes` as it requires very specific knowledge about what you are trying to do.

File: frontend/src/app.d.ts (hunk 1 above)
Comment (myieye) [reply frontend/src/app.d.ts:14]: That works for me.

File: frontend/src/hooks.shared.ts (hunk 2 above)
Comment (myieye) [reply frontend/src/hooks.shared.ts]: Made the change

**Review by hahn-kev — APPROVED** [review]
looks good!

**Pull request merged**